lib/hanami/cli/generators/app/slice.rb

```diff
@@ -14,23 +14,24 @@ def initialize(fs:, inflector:)
             @inflector = inflector
           end
 
-          def call(app, slice, slice_url_prefix, context: SliceContext.new(inflector, app, slice, slice_url_prefix)) # rubocop:disable Metrics/AbcSize
-            fs.inject_line_before_last(fs.join("config", "routes.rb"), /end/, t("routes.erb", context).chomp)
+          def call(app, slice, slice_url_prefix, context: SliceContext.new(inflector, app, slice, slice_url_prefix))
+            fs.inject_line_at_block_bottom(fs.join("config", "routes.rb"), /define/, t("routes.erb", context).chomp)
 
             fs.mkdir(directory = "slices/#{slice}")
 
             fs.chdir(directory) do
+              fs.write("slice.rb", t("slice.erb", context))
```

Comment:
This actually needs to go into `config/slices/[slice_name].rb` instead of `slices/[slice_name]/slice.rb`.

Reply:
And technically, having this file in the first place is completely optional — we don't _need_ the concrete class definition to exist (the framework will define the class for us if it doesn't), but I agree that it probably makes more sense to generate the file than not to. The only downside is that the app may end up feeling a little "heavier" because we're adding files that we don't technically need.

Reply:
@timriley If not strictly needed, I'd be conservative here and NOT generate for now.

We can always add it to the gen files, but it's harder to remove features.